tw/debug/ScavengingOverview.js

```diff
@@ -0,0 +1,80 @@
+/**** game/Modules/Overview/ScavengingOverview.js ****/
+define("Ig/TribalWars/Modules/Overview/ScavengingOverview", ["Ig/TribalWars/Modules/Scavenging/ScavengeOption", "Ig/TribalWars/Modules/Scavenging/ScavengingService", "Ig/TribalWars/Modules/Scavenging/ScavengeVillage", "Ig/TribalWars/Modules/VillagePlace/ScavengeScreen/DialogLauncher", "Ig/TribalWars/Modules/Scavenging/ScavengeVillageSynchronizer"], function(o, t, i, r, n) {
+    "use strict";
+
+    function e(n, e) {
+        for (var a in this.option_bases = n, this.scavenging_service = new t, this.village_data = {}, e) e.hasOwnProperty(a) && (this.village_data[a] = i.createFromDTO(this.option_bases, e[a]))
+    }
+    return e.prototype = {
+        init: function() {
+            var i = this;
+            $.each(i.village_data, function(n, t) {
+                i.updateVillageStatus(t), $.each(t.options, function(n, e) {
+                    var a = t;
+                    e.on(o.EVENT_STATUS_CHANGED, function() {
+                        i.updateVillageStatus(a)
+                    })
+                })
+            }), UI.ToolTip(".scavenging-row .scavenging-option", {
+                bodyHandler: function() {
+                    var n = $(this),
+                        e = n.parents(".scavenging-row"),
+                        a = i.village_data[e.data("village")],
+                        t = n.data("option");
+                    return i.getTooltipBody(a, t)
+                }
+            }), $(".scavenging-row .scavenging-option").on("mouseover", function() {
+                Timing.tickHandlers.timers.initTimers("scavenging-timer", null)
+            }).on("click", function() {
+                return i.handleClick($(this)), !1
+            }), this.village_synchronizer = new n(this.village_data, this.scavenging_service), $(TribalWars).on("global_tick", function() {
+                i.village_synchronizer.processTiming()
+            })
+        },
+        handleClick: function(n) {
+            var e = n.parents(".scavenging-row").data("village"),
+                a = n.data("option"),
+                t = this.village_data[e],
+                i = t.options[a];
+            i.getStatus() === o.STATUS_LOCKED ? new r(t, this.scavenging_service, null, null).openUnlockOptionDialog(a) : i.getStatus() === o.STATUS_INACTIVE && Dialog.fetch("scavenge", "place", {
```

Comment:
```
new r(t, this.scavenging_service, null, null)
```
DialogLauncher is constructed with null rate_schedule and amount_schedule

This part should be an easy fix.  Check out ResourcesForecaster.fetchSchedules

e.g. 
```
ResourcesForecaster.fetchSchedules(t.id, schedules => {
    new r(t, this.scavenging_service, schedules.rates, schedules.amounts)
        .openUnlockOptionDialog(a)
})
```